### PR TITLE
Fix emitting debug information for structe types with `SCALANATIVE_OPTIMIZE=false`

### PR DESCRIPTION
**Pull request by tanishiking**
fix https://github.com/scala-native/scala-native/issues/3435

Now, the following Scala code without optimization will emit

```scala
import scala.scalanative.runtime.struct

object Main {
  @struct class SC1(val b: Int, val c: Boolean)

  def main(args: Array[String]): Unit = {
    val x = new SC1(5, true)
    println(x.b)
  }
}
```

```
!22 = !DIBasicType(name: "int", size: 32, align: 32, encoding: DW_ATE_signed)
!23 = !DIDerivedType(tag: DW_TAG_member, baseType: !22, size: 32)
!24 = !DIBasicType(name: "bool", size: 8, align: 8, encoding: DW_ATE_boolean)
!25 = !DIDerivedType(tag: DW_TAG_member, baseType: !24, size: 8)
!26 = !{!23, !25}
!27 = !DICompositeType(tag: DW_TAG_structure_type, size: 64, elements: !26)
```

Since we don't have `name` attributs in `DICompositeType` and `DIBasicType`, we will debugger can't show their type name and members' names.
In order to preserve those information, we need to somehow change the way of traversing NIR or adding name information into NIR

```
(__lldb_invalid_typedef_name) oldCar = {
   = 1980
   = false
   = 0xb96d000190b47e3c
}
(__lldb_invalid_typedef_name) myCar1 = {
   = 16260
   = true
   = 0x0000000100009c40
}
```

### CODE REVIEW

**Comment by tanishiking** [comment]
Due to the issue https://github.com/scala-native/scala-native/issues/3442 this PR is based on `36d32f0`

File: tools/src/main/scala/scala/scalanative/codegen/llvm/AbstractCodeGen.scala [hunk 1]
```diff
@@ -17,6 +17,8 @@ import scala.scalanative.codegen.{Metadata => CodeGenMetadata}
 
 import scala.language.implicitConversions
 import scala.scalanative.codegen.llvm.Metadata.conversions._
+import scala.scalanative.nir.Global.Member
+import scala.scalanative.nir.Global.Top
```
Comment (WojciechMazur) [inline tools/src/main/scala/scala/scalanative/codegen/llvm/AbstractCodeGen.scala:21]: Spourious change

**Review by WojciechMazur — APPROVED** [review]
Overall looks good. We potentially can use indexes if we would stumble upon unnamed structs, but overall it would be better to have an actual names in nir. Currently most of the structs are used for intrinsic stuff (eg. generated values). We potentially can use `scalanative.linker.Class` / `scalanative.codegen.FieldLayour` for the user defined types which are also struct types, but let's leave it for another dedicated PR

**Comment by tanishiking** [comment]
> We potentially can use indexes if we would stumble upon unnamed structs, but overall it would be better to have an actual names in nir.

Maybe it's better to add field name info into `StructValue` https://github.com/tanishiking/scala-native/blob/f0b390e4dacdefa74c04a8f663f93f6402ae70a5/nir/src/main/scala/scala/scalanative/nir/Types.scala#L70

like 

```
final case class StructValue(
  name: String, // name of this struct, SC1 in the example above,
  tys: Seq[(String, Type)] // tuple of member name and type
) extends AggregateKind
```

before release of 0.5.0?

anyway, we can work on this in another PR :+1:

**Comment by WojciechMazur** [comment]
> Maybe it's better to add field name info into StructValue 

That's one of solutions, but I don't think we should mix metadata with actual values. Also, the @struct is one of the remaining special use cases just to be able to use functions which return struct by value. All other StructValues are completly internal, and emited in the codegen, in most cases they represent the field layout of a class. 
Also, in debugger we would rather be interested in the local variable of `x: SC1`. When it comes to the composite type describing `SC1` we can even make it up by ourselfs my mangling the types and assuming this is an anonymous struct (which is true for all `CStructN` types currently.

**Pull request merged**